### ISSUE TITLE
RFE: Support for "merge mode" in external repos

### ISSUE THREAD

**Issue by ignatenkobrain** [ktdreyer/koji-ansible#127]
Baiscally it is not possible to set up el8 targets which would like to use plain el8 repos...

```
❯ koji -p gdc list-external-repos --tag tools-el8-build
Pri External repo name        Mode       URL
--- ------------------------- ---------- ----------------------------------------
5   centos8                   bare       https://pulp1.na.intgdc.com/pulp/repos/repos/centos/81/testing/$arch/baseos/
7   centos8-appstream         bare       https://pulp1.na.intgdc.com/pulp/repos/repos/centos/81/testing/$arch/appstream/
8   centos8-powertools        bare       https://pulp1.na.intgdc.com/pulp/repos/repos/centos/81/testing/$arch/powertools/
10  epel8                     bare       https://pulp1.na.intgdc.com/pulp/repos/centos/epel/testing/8/$arch/
```

this is, for instance, what we have in our koji instance. Setting `bare` mode is MUST.

Could you add an option to override it? Thanks!

**Comment by ktdreyer** [comment]
You're right. I am tracking this in https://github.com/ktdreyer/koji-ansible/issues/36

I pushed a work-in-progress branch named `external-repos` just now, and you can test it out. It has a prerequisite commit that I'm going to submit in a separate PR ("koji_tag: remove then add in ensure_external_repos") after more testing.

**Comment by ktdreyer** [comment]
I rebased `external-repos` now to pick up today's latest changes in `master`.

**Comment by ignatenkobrain** [comment]
I have been using that branch for a while and it works quite well here!

**Comment by ignatenkobrain** [comment]
@ktdreyer what's missing to get that branch merged?

**Comment by ignatenkobrain** [comment]
ping @ktdreyer

**Comment by ktdreyer** [comment]
Sorry this is taking a while. I rebased `external-repos` on to the latest master again today.

The reason I have not merged `external-repos` to master yet is that I need to do more testing. In particular, I want to be completely sure that the changes will not to break the existing koji-ansible users.

I ran this branch in check mode on some large playbooks, and I did not see any changes, but I still want to have a few more days of testing. We have CI tests, but I'm not sure they cover everything that this branch changes yet.

My plan is to break this branch up into individual PRs. "koji_tag: remove then add in ensure_external_repos" should land first, and then we'll take the other commit "koji_tag: add merge_mode support for external repos".

It's reasonable to expect this in master some time next week.

**Comment by ignatenkobrain** [comment]
Thanks! We are extensively using these modules and it is complicated to tell everybody to just clone git repo, checkout some branch and build collection themselves :)

**Comment by ktdreyer** [comment]
Thanks for letting me know. I will start merging the first parts of this branch next week.

I see that we don't cover every line in `koji_tag.ensure_external_repos()` yet, so I'm going to add unit tests to fully cover that method, and then post the pull requests to update it with the `external-tests` changes.

**Comment by ktdreyer** [comment]
I've added more tests, and I rebased `external-repos` on to `master` again today.

PR #149 includes "koji_tag: remove then add in ensure_external_repos". Once we merge that, I'll post the rest of the `external-repos` branch as a second PR.

**Comment by ktdreyer** [comment]
I've rebased `external-repos` on to `master` again today.

PR #151 includes the final changes from this branch.
